### PR TITLE
implement Deny card

### PR DESCRIPTION
**Pull request by ThunderDev1**

### CODE REVIEW

**Comment by ThunderDev1** [comment]
Closes #26

**Comment by hissalht** [comment]
image
🤔

File: app/src/bf-game/index.js [hunk 1]
```diff
@@ -165,26 +165,82 @@ export function drawCard(gameState) {
...
+  } else {
+    // we need to save this info in case a loot card is played, followed by
+    // two deny cards, cancelling each other out, so we know the looter id
+    card.userId = userId;
+    newGameState.discardPile.unshift(card);
+  }
+
+  if (card.type === CardType.Deny) {
+    // force timer reset
+    newGameState.timerRandom = Date.now();
```
Comment (hissalht) [inline app/src/bf-game/index.js:204]: Document `GameState.timerRandom` on the type definition jsdoc

File: app/src/bf-game/index.js [hunk 2]
```diff
@@ -165,26 +165,82 @@ export function drawCard(gameState) {
...
+}
+
+/**
+ * Calculate the new state of the game when a player chooses a card to play.
+ *
+ * @param {GameState} gameState Current state of the game
+ * @param {Card} priorityCard A card that can skip submit phase (eg. Resurect)
+ * @returns {GameState} New state of the game.
+ */
+export function playCard(gameState, userId, priorityCard) {
```
Comment (hissalht) [inline app/src/bf-game/index.js:217]: `priorityCard` is not documented

File: app/src/views/Game.vue [hunk 3]
```diff
@@ -226,6 +238,21 @@ export default {
 
         this.peekCountDown = setInterval(timer, 1000);
       }
+    },
+    isSubmitting(newValue, oldValue) {
+      console.log("is submitting", newValue);
+      if (!newValue) clearInterval(this.submitCountDown);
+      if (newValue && newValue !== oldValue) {
```
Comment (hissalht) [inline app/src/views/Game.vue]: Is this check necessary ?

**Review by hissalht — APPROVED** [review]

File: app/src/bf-game/index.js (hunk 1 above)
Comment (ThunderDev1) [reply app/src/bf-game/index.js:204]: It's to indicate the timer needs to be reset, seems like a hack, but didn't find a better way for now..

File: app/src/bf-game/index.js (hunk 2 above)
Comment (ThunderDev1) [reply app/src/bf-game/index.js:217]: Hmmm, it is documented line 213, is there something I'm missing ?

File: app/src/views/Game.vue (hunk 3 above)
Comment (ThunderDev1) [reply app/src/views/Game.vue]: No, I refactored the watcher for more clarity.

**Pull request merged**